src/runtime_src/core/edge/drm/zocl/zocl_bo.c

```diff
@@ -967,17 +1021,36 @@ void zocl_init_mem(struct drm_zocl_dev *zdev, struct mem_topology *mtopo)
 		/* In mem_topology, size is in KB */
 		memp->zm_size = md->m_size * 1024;
 		memp->zm_used = 1;
+		INIT_LIST_HEAD(&memp->zm_mm_list);
 
-		if (!strstr(md->m_tag, "MIG") && !strstr(md->m_tag, "LPDDR")) {
+		if (strstr(md->m_tag, "DDR") && (memp->zm_base_addr == 0x0)) {
 			memp->zm_type = ZOCL_MEM_TYPE_CMA;
 			continue;
 		}
 
 		memp->zm_mm = vzalloc(sizeof(struct drm_mm));
-		memp->zm_type = ZOCL_MEM_TYPE_PL_LP_DDR;
-
+		memp->zm_type = ZOCL_MEM_TYPE_RANGE_ALLOC;
+	
 		drm_mm_init(memp->zm_mm, memp->zm_base_addr, memp->zm_size);
 	}
+
+	/* Create a link list for similar memory manager */
+	for (i = 0; i < zdev->num_mem; i++) {
+		memp = &zdev->mem[i];
+		if (!memp->zm_used)
+			continue;
+
+		for (j = 0; j < zdev->num_mem; j++) {
+			if ((i == j) || !mtopo->m_mem_data[j].m_used)
+				continue;
+
+			if (strstr(mtopo->m_mem_data[i].m_tag, mtopo->m_mem_data[j].m_tag) && 
```

Comment:
strstr just looks for substrings. We need full comparision. Use strcmp

Reply:
Updated.